src/common/mem.c

```diff
@@ -338,44 +356,20 @@ umem_tx_errno(int err)
 
 /* volatile memroy operations */
 
-static umem_id_t
-vmem_id(const struct umem_instance *umm, void *addr)
-{
-	umem_id_t	ummid = UMMID_NULL;
-
-	ummid.off = (uint64_t)addr;
-	return ummid;
-}
-
-static void *
-vmem_addr(const struct umem_instance *umm, umem_id_t ummid)
-{
-	return (void *)ummid.off;
-}
-
-static bool
-vmem_equal(struct umem_instance *umm, umem_id_t ummid1, umem_id_t ummid2)
-{
-	return ummid1.off == ummid2.off;
-}
-
 static int
-vmem_free(struct umem_instance *umm, umem_id_t ummid)
+vmem_free(struct umem_instance *umm, umem_off_t umoff)
 {
-	if (ummid.off != 0)
-		free((void *)ummid.off);
+	free(umem_off2ptr(umm, umoff));
+
 	return 0;
 }
 
-umem_id_t
+umem_off_t
 vmem_alloc(struct umem_instance *umm, size_t size, uint64_t flags,
 	   unsigned int type_num)
 {
-	umem_id_t ummid = UMMID_NULL;
-
-	ummid.off = (uint64_t)(flags & POBJ_FLAG_ZERO ?
-			       calloc(1, size) : malloc(size));
-	return ummid;
+	return (uint64_t)((flags & POBJ_FLAG_ZERO) ?
```

Comment:
nit - I know umem_off_t is a uint64_t, but for consistency, can you make the return type and the cast type the same? :)

Reply:
yeah, fair point.   I think it's just cause I copied it from the old code :-)